OBSDOCS-1423: Document that instrumentation injection is GA (#84370)
openshift/openshift-docs: modules/otel-config-instrumentation.adoc

|===

[id="otel-configuration-of-opentelemetry-sdk-variables_{context}"]
== Configuration of the OpenTelemetry SDK variables

You can use the `instrumentation.opentelemetry.io/inject-sdk` annotation in the OpenTelemetry Collector custom resource to instruct the {OTELName} Operator to inject the following OpenTelemetry SDK environment variables into your pod:

* `OTEL_SERVICE_NAME`
* `OTEL_TRACES_SAMPLER`
* `OTEL_TRACES_SAMPLER_ARG`
* `OTEL_PROPAGATORS`
* `OTEL_RESOURCE_ATTRIBUTES`
* `OTEL_EXPORTER_OTLP_ENDPOINT`
* `OTEL_EXPORTER_OTLP_CERTIFICATE`
* `OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE`
* `OTEL_EXPORTER_OTLP_CLIENT_KEY`

.Values for the `instrumentation.opentelemetry.io/inject-sdk` annotation
[cols="a, a"]
|===
| Value | Description

> |`"true"`

Reviewer: The instrumentation CR name is also considered in the annotation. 

**"true" - to inject Instrumentation resource with default name from the current namespace.**
"my-instrumentation" - to inject Instrumentation CR instance with name "my-instrumentation" in the current namespace.
"my-other-namespace/my-instrumentation" - to inject Instrumentation CR instance with name "my-instrumentation" from another namespace "my-other-namespace".
"false" - do not inject

Author: Thank you, Ishwar. It's on the next several lines.

|Injects the `+Instrumentation+` resource from the current namespace.

|`"false"`
|Injects no `+Instrumentation+` resource.

|`"<instrumentation_name>"`
|Specifies the name of the `+Instrumentation+` resource to inject from the current namespace.

|`"<namespace>/<instrumentation_name>"`
|Specifies the name of the `+Instrumentation+` resource to inject from another namespace.

|===

[id="otel-configuration-of-apache-http-server-auto-instrumentation_{context}"]
== Configuration of the Apache HTTP Server auto-instrumentation

> include::snippets/technology-preview.adoc[leveloffset=+1]

Author: ```suggestion
:FeatureName: The Apache HTTP Server auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]
```

|===

[id="otel-configuration-of-opentelemetry-sdk-variables_{context}"]
== Configuration of the OpenTelemetry SDK variables

> You can use the `instrumentation.opentelemetry.io/inject-sdk` annotation in the OpenTelemetry Collector custom resource to instruct the {OTELName} Operator to inject the following OpenTelemetry SDK environment variables into your pod:

Reviewer 2: to inject some of the following OpenTelemetry SDK environment variables depending on the Instrumentation CR into your pod.

The reason being not all of these env vars are injected, e.g. if TLS is not specified the  OTEL_EXPORTER_OTLP_CERTIFICATE (and others) are not set

Author: Thank you. Applied.

* `OTEL_SERVICE_NAME`
* `OTEL_TRACES_SAMPLER`
* `OTEL_TRACES_SAMPLER_ARG`
* `OTEL_PROPAGATORS`
* `OTEL_RESOURCE_ATTRIBUTES`
* `OTEL_EXPORTER_OTLP_ENDPOINT`
* `OTEL_EXPORTER_OTLP_CERTIFICATE`
* `OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE`
* `OTEL_EXPORTER_OTLP_CLIENT_KEY`

.Values for the `instrumentation.opentelemetry.io/inject-sdk` annotation
[cols="a, a"]
|===
| Value | Description

|`"true"`
|Injects the `+Instrumentation+` resource with the default name from the current namespace.

|`"false"`

> |Injects no `+Instrumentation+` resource.

Reviewer 3: It means that this setting does not inject `+Instrumentation+`, right? 

Can it be revised something like that?

|===

[id="otel-configuration-of-opentelemetry-sdk-variables_{context}"]
== Configuration of the OpenTelemetry SDK variables

> You can use the `instrumentation.opentelemetry.io/inject-sdk` annotation in the OpenTelemetry Collector custom resourceto instruct the {OTELName} Operator to inject the following OpenTelemetry SDK environment variables into your pod:

Author: ```suggestion
You can use the `instrumentation.opentelemetry.io/inject-sdk` annotation in the OpenTelemetry Collector custom resource to instruct the {OTELName} Operator to inject the following OpenTelemetry SDK environment variables into your pod:
```

[id="otel-configuration-of-dotnet-auto-instrumentation_{context}"]
== Configuration of the .NET auto-instrumentation

> include::snippets/technology-preview.adoc[leveloffset=+1]

Author: ```suggestion
:FeatureName: The .NET auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]
```

[IMPORTANT]
====

> This feature by default injects not supported upstream instrumentation libraries.

Author: ```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

Author: Applied.

[id="otel-configuration-of-go-auto-instrumentation_{context}"]
== Configuration of the Go auto-instrumentation

> include::snippets/technology-preview.adoc[leveloffset=+1]

Author: ```suggestion
:FeatureName: The Go auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]
```

[IMPORTANT]
====

> This feature by default injects not supported upstream instrumentation libraries.

Author: ```suggestion
By default, this feature injects unsupported, upstream instrumentation libraries.
```

> This feature by default injects not supported upstream instrumentation libraries.

Author: Thank you for the suggestion. Fixed.

[id="otel-configuration-of-go-auto-instrumentation_{context}"]
== Configuration of the Go auto-instrumentation

:FeatureName: The Go auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]

[IMPORTANT]
====

> By default, this feature injects unsupported, upstream instrumentation libraries.

Reviewer 3: ```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

I don't think comma is required here. Hence I suggest to remove that.

Author: Thank you for the suggestion. Using a comma here is correct in my understanding of punctuation.

[id="otel-configuration-of-java-auto-instrumentation_{context}"]
== Configuration of the Java auto-instrumentation

> include::snippets/technology-preview.adoc[leveloffset=+1]

Author: Thank you so much for spotting this. Fixing.

> include::snippets/technology-preview.adoc[leveloffset=+1]

Author: ```suggestion
:FeatureName: The Java auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]
```

[IMPORTANT]
====

> This feature by default injects not supported upstream instrumentation libraries.

Author: ```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

> This feature by default injects not supported upstream instrumentation libraries.

Author: Thank you for the suggestion. Fixed.

[id="otel-configuration-of-java-auto-instrumentation_{context}"]
== Configuration of the Java auto-instrumentation

:FeatureName: The Java auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]

[IMPORTANT]
====

> By default, this feature injects unsupported, upstream instrumentation libraries.

Reviewer 3: ```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

I don't think comma is required here. Hence I suggest to remove that.

[id="otel-configuration-of-nodejs-auto-instrumentation_{context}"]
== Configuration of the Node.js auto-instrumentation

> include::snippets/technology-preview.adoc[leveloffset=+1]

Author: ```suggestion
:FeatureName: The Node.js auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]
```

[IMPORTANT]
====

> This feature by default injects not supported upstream instrumentation libraries.

Author: ```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

> This feature by default injects not supported upstream instrumentation libraries.

Author: Thank you for the suggestion. Fixed.

[id="otel-configuration-of-nodejs-auto-instrumentation_{context}"]
== Configuration of the Node.js auto-instrumentation

:FeatureName: The Node.js auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]

[IMPORTANT]
====

> By default, this feature injects unsupported, upstream instrumentation libraries.

Reviewer 3: ```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

I don't think comma is required here. Hence I suggest to remove that.

Author: Thank you for the suggestion. Using a comma here is correct in my understanding of punctuation.

[id="otel-configuration-of-python-auto-instrumentation_{context}"]
== Configuration of the Python auto-instrumentation

> include::snippets/technology-preview.adoc[leveloffset=+1]

Author: ```suggestion
:FeatureName: The Python auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]
```

[IMPORTANT]
====

> This feature by default injects not supported upstream instrumentation libraries.

Author: ```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

> This feature by default injects not supported upstream instrumentation libraries.

Author: Thank you for the suggestion. Fixed.

[id="otel-configuration-of-python-auto-instrumentation_{context}"]
== Configuration of the Python auto-instrumentation

:FeatureName: The Python auto-instrumentation
include::snippets/technology-preview.adoc[leveloffset=+1]

[IMPORTANT]
====

> By default, this feature injects unsupported, upstream instrumentation libraries.

Reviewer 3: Same comment

```suggestion
By default, this feature injects unsupported upstream instrumentation libraries.
```

Author: Thank you for the suggestion. Using a comma here is correct in my understanding of punctuation.
